ipaserver/install/cainstance.py

```diff
@@ -1711,6 +1711,63 @@ def make_entry(cert, entry):
     return __update_entry_from_cert(make_filter, make_entry, cert)
 
 
+def get_ca_renewal_nickname(cert):
+    """
+    Get the nickname for storage in the cn_renewal container.
+
+    Assumes that the certificate is signed (possibly self-signed) by
+    the IPA CA.  Therefore DO NOT use this function for adding an
+    externally-signed CA certificate.
+
+    :param cert: A python-cryptography X509Certificate
+    :return: string, or None if nickname cannot be determined.
+
+    """
+    idn = DN(cert.issuer)
+    sdn = DN(cert.subject)
+
+    nickname_by_prefix = {
+        RDN('CN=CA Audit'): 'auditSigningCert cert-pki-ca',
+        RDN('CN=OCSP Subsystem'): 'ocspSigningCert cert-pki-ca',
+        RDN('CN=CA Subsystem'): 'subsystemCert cert-pki-ca',
+        RDN('CN=KRA Audit'): 'auditSigningCert cert-pki-kra',
+        RDN('CN=KRA Transport Certificate'): 'transportCert cert-pki-kra',
+        RDN('CN=KRA Storage Certificate'): 'storageCert cert-pki-kra',
+        RDN('CN=IPA RA'): 'ipaCert',
+    }
+
+    if idn == sdn:
+        return 'caSigningCert cert-pki-ca'
+    else:
+        return nickname_by_prefix.get(sdn[0])
+
+
+def update_ca_renewal_entry(conn, nickname, cert):
+    """
+    Update the ca_renewal entry for the given nickname.
+
+    :param conn: A *connected* LDAP handle
+    :param nickname: NSSDB nickname
+    :param cert: python-cryptography X509Certificate
+
+    """
+    dn = DN(('cn', nickname), ('cn', 'ca_renewal'),
```

Comment:
Please add the base dn for ca_renewal to ``ipalib/constants.py`` and refer to it as ``DN(('cn', nickname), api.env.container_ca_renewal, api.env.basedn)``.